packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -268,6 +274,39 @@ function dedupeTransactions(txs: Transfer[]) {
   )
 }
 
+export async function fetchBatchedWithdrawals(
+  params: FetchWithdrawalsParams & {
+    batchSizeBlocks?: number
+  }
+): Promise<Withdrawal[]> {
+  const latestBlockNumber = await params.l2Provider.getBlockNumber()
+
+  const fromBlock = params.fromBlock ?? 1
+  const toBlock = params.toBlock ?? latestBlockNumber
+
+  const batchSizeBlocks = params.batchSizeBlocks ?? 5_000_000
+  const batchCount = Math.ceil((toBlock - fromBlock) / batchSizeBlocks)
+
+  const promises = Array.from({ length: batchCount }, (_, i) => {
```

Comment:
Maybe we can limit the number of fetch in parallel? If we have a low batchCount we might 429 error

Reply:
I don't think we need to overcomplicate it, because this will be replaced by the indexer soon enough